dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -13,10 +13,21 @@ def update_last_confirmation_date
   def update
     return unless school_info_params[:school_id].present? || school_info_params[:country].present?
 
+    new_school_info_params =
+      if school_info_params[:full_address]&.blank?
+        school_info_params.except(:full_address)
+      else
+        school_info_params
+      end
+
+    if new_school_info_params[:country]&.downcase&.eql? 'united states'
```

Comment:
Should this (at least theoretically) be generalized to all countries? Should "France" really be "FR" for instance?

It's very possible we don't really care for non-US locales and that this is at least a good short-medium term fix. I'm more asking for my own curiosity.

Reply:
Good question!  My guess is that it would be generalized to all countries as it will be standardized and in line with how we identify different locales, but I am not sure.  When I get additional information I will update.

Reply:
We could set up a reverse mapping using the constants in this file: https://github.com/code-dot-org/code-dot-org/blob/staging/lib/country_codes.rb